File: RedSheet.DbEntities/Models/vNanoOurBatnaRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vNanoOurBatnaRecords", Schema = "dbo")]
	[Serializable]
	public partial class vNanoOurBatnaRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NanoOurBatnaId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string NanoOurBatnaValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 

	public vNanoOurBatnaRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-nano-our-batna-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vNanoOurBatnaRecord {
    constructor(vNanoOurBatnaRecord?: vNanoOurBatnaRecord )  {
        let properties = [ "nanoOurBatnaId", "nanoOurBatnaValue", "projectModuleId",];
        for (let property of properties)
            if (vNanoOurBatnaRecord && vNanoOurBatnaRecord[property])
                this[property] = vNanoOurBatnaRecord[property];
    }
 
	nanoOurBatnaId : number =   0 ;
 
    @required()
    @maxLength(1000)
	nanoOurBatnaValue : string =   undefined;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;


}
